### ISSUE TITLE
[$250] [HOLD for Payment 2024-09-06][Search v2.1] Include policyID in the Search AST

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#46592]
## Problem
Currently, we send the policyIDs as a Search API param, however it should be parsed an included in the AST, like we do for other filters.

## Solution
Include `policyID` in the parser grammar and stop sending the `policyIDs` param

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021833303422059505984</li>
        <li>Upwork Job ID: 1833303422059505984</li>
        <li>Last Price Increase: 2024-09-10</li>
    </ul>
</details>

**Comment by mallenexpensify** [comment]
> @luacmartins Is this waiting on backend changes?

Same question as @rayane-djouah , I'm unsure of next steps (and commenting to remove `Overdue` )

**Comment by luacmartins** [comment]
I think @Kicu might work on this one. Can you confirm?

**Comment by Kicu** [comment]
yes! I'm working, thanks for pointing out this issue, I missed it

**Comment by Kicu** [comment]
@luacmartins draft is ready so you might want to take a look. This is surprisingly complex, because I had to make sure all the different cases where we might set `policyID` from WorkspaceSwitcher are handled.
That is why the changes are spread over multiple files.

Tomorrow I will polish this up and finish the checklist

**Comment by luacmartins** [comment]
Nice! Thanks for linking the draft. I'll take a look at it today.

**Comment by Kicu** [comment]
one thing I forgot to mention - I believe now that `policyID: xxxx` is sent on top level of AST query I think backend is not correctly filtering the results.
Do you mind checking that out @luacmartins ?

**Comment by luacmartins** [comment]
I saw that last night when working on another PR. Yea, I'll work on this today.

**Comment by luacmartins** [comment]
Bump for payment

**Comment by mallenexpensify** [comment]
@rayane-djouah can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~021833303422059505984

I'm assuming this issue is $250.

**Comment by mallenexpensify** [comment]
@rayane-djouah , I think I accidentally sent you an invite without hiring you, can you please accept the job and reply here once you have?  
https://www.upwork.com/jobs/~021833303422059505984

**Comment by mallenexpensify** [comment]
Contributor+: @rayane-djouah paid $250 via Upwork

I'm assuming we don't need a regression test here and that they'll be created as part of the Search project. Thx!
